### PR TITLE
Bug 26819 - Bounded DNLC cache unitTested skipBuild review @ Satya

### PR DESCRIPTION
**Pull request by skosuru-mapr**

### CODE REVIEW

File: src/cache_inode/cache_inode_init.c [hunk 1]
```diff
@@ -50,6 +50,19 @@
  * @return CACHE_INODE_SUCCESS or errors.
  *
  */
+
+extern uint64_t number_of_dirent_entries;
+void*
+dirent_entries_monitor(void *notused)
+{
+  while(1) {
+    if(number_of_dirent_entries > 1000*1000) {
```
Comment (satgs) [inline src/cache_inode/cache_inode_init.c]: Increase it much larger value. Do we have an estimate on how much memory will a million entries take.

File: src/cache_inode/cache_inode_misc.c [hunk 2]
```diff
@@ -812,6 +814,7 @@ cache_inode_release_dirents(cache_entry_t *entry,
 			if (dirent->ckey.kv.len)
 				cache_inode_key_delete(&dirent->ckey);
 			gsh_free(dirent);
+      number_of_dirent_entries--;
```
Comment (satgs) [inline src/cache_inode/cache_inode_misc.c]: Fix indentation. Sent you an email on how your git should be configured. Also, run githooks script.

File: src/cache_inode/cache_inode_readdir.c [hunk 3]
```diff
@@ -223,6 +225,7 @@ cache_inode_operate_cached_dirent(cache_entry_t *directory,
 			/* try to rename--no longer in-place */
 			dirent3 = gsh_malloc(sizeof(cache_inode_dir_entry_t)
 					     + newnamesize);
+      number_of_dirent_entries++;
```
Comment (satgs) [inline src/cache_inode/cache_inode_readdir.c]: Check this. We are doing rename. Will old entry be deleted ? If so, does incrementing make sense ?
Also check if you would have to decrement the value in CACHE_INODE_DIRENT_OP_REMOVE.

**Review by satgs — COMMENTED** [review]
Please address the comments.

File: src/cache_inode/cache_inode_readdir.c (hunk 3 above)
Comment (skosuru-mapr) [reply src/cache_inode/cache_inode_readdir.c]: In CACHE_INODE_DIRENT_OP_REMOVE case we are not really removing an entry. I decrementing the count exactly, where we are removing the entry. In the case of rename also I am incrementing the counter, as entry is allocated.

File: src/cache_inode/cache_inode_init.c (hunk 1 above)
Comment (skosuru-mapr) [reply src/cache_inode/cache_inode_init.c]: each entry is taking around 100 bytes, I am changing this to 100 million.

File: src/cache_inode/cache_inode_misc.c (hunk 2 above)
Comment (skosuru-mapr) [reply src/cache_inode/cache_inode_misc.c]: I ran the script and fixed indentation.

**Review by satgs — APPROVED** [review]
1. Add a configuration param for the max-entries.
2. Look into avl_tree_set_deleted code. Try to understand what delete is doing.

Looks good for now but address the above.

**Pull request merged**